blackmarket/UI/ViewModels/SignUpViewModel.swift

```diff
@@ -78,8 +80,17 @@ class SignUpViewModel: ObservableObject, Identifiable {
         }
     }()
     
-    func logIn() {
-        print("Log in button tapped")
+    init(authServices: AuthenticationServices = AuthenticationServices()) {
+        self.authServices = authServices
+    }
+    
+    func signUp() async {
+        do {
+            _ = try await authServices.signup(email: emailConfiguration.value,name: nameConfiguration.value, password: passwordConfiguration.value)
```

Comment:
the @discardableResult wrapper shoulld be applied to the function you are calling without using the output, in this case is `autheServices.signup`

Reply:
This can now be:
```suggestion
            try await authServices.signup(email: emailConfiguration.value,name: nameConfiguration.value, password: passwordConfiguration.value)
```